codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -526,32 +529,29 @@ private boolean exceedsMaxHeaderSize(long size) {
         return true;
     }
 
-    private byte[] readStringLiteral(InputStream in, int length) throws IOException {
-        byte[] buf = new byte[length];
-        if (in.read(buf) != length) {
-            throw DECOMPRESSION_EXCEPTION;
-        }
-
+    private CharSequence readStringLiteral(ByteBuf in, int length) throws IOException {
         if (huffmanEncoded) {
-            return Huffman.DECODER.decode(buf);
+            return huffmanDecoder.decode(in, length);
         } else {
-            return buf;
+            byte[] buf = new byte[length];
+            in.readBytes(buf);
+            return new AsciiString(buf, false);
```

Comment:
@carl-mastrangelo I think it is fine as we state in the package docs that it is a partly "compliant" hpack impl that is only ok when used with http2.